k8sfv/scale_test.go

```diff
@@ -160,6 +164,49 @@ var _ = Context("with a k8s clientset", func() {
 			d = NewDeployment(clientset, 0, true)
 		})
 
+		It("denied packets between 2 local endpoints", func() {
+			for cycle, isolated := range []bool{false, true, false, true, false, true} {
+				nsName := nsPrefix + "test" + fmt.Sprintf("%v", cycle)
+				if isolated {
+					createIsolatedNamespace(clientset, nsName, nil)
+				} else {
+					createNamespace(clientset, nsName, nil)
+				}
+				pod1 := createPod(clientset, d, nsName, podSpec{})
+				pod2 := createPod(clientset, d, nsName, podSpec{})
+				for ii := 0; ii < 5; ii++ {
+					// Run an nmap scan between the pods.
+					runNmap(pod1, pod2)
+					sumCalicoDeniedPackets(felixIP)
+
+					// Get Felix to GC and dump heap memory profile.
+					exec.Command("pkill", "-USR1", "calico-felix").Run()
+					time.Sleep(2 * time.Second)
```

Comment:
Bit fiddly, as I'd have to have the same logic as Felix for the file name (template + timestamp).